bagit.py

```diff
@@ -182,34 +214,73 @@ def make_bag(bag_dir, bag_info=None, processes=1, checksums=None, checksum=None,
+            if dest_dir:
+                temp_data = tempfile.mkdtemp(dir=dest_dir)
+                data_dir = os.path.join(dest_dir, 'data')
+                data_subdir = os.path.relpath(
+                    bag_dir, os.path.join(bag_dir, os.pardir)
+                )
+                if os.path.abspath(os.path.join(bag_dir, data_subdir)) == bag_dir:
+                    raise RuntimeError(
+                        _("Could not get parent of source directory %s") % bag_dir
+                    )
+            else:
+                temp_data = tempfile.mkdtemp(dir=bag_dir)
+                data_dir = os.path.join(bag_dir, 'data')
 
-            for f in os.listdir('.'):
-                if os.path.abspath(f) == temp_data:
+            for f in os.listdir(bag_dir):
+                if os.path.join(bag_dir, f) == temp_data:
                     continue
                 new_f = os.path.join(temp_data, f)
-                LOGGER.info(_('Moving %(source)s to %(destination)s'),
-                            {'source': f, 'destination': new_f})
-                os.rename(f, new_f)
+                if dest_dir:
+                    LOGGER.info(_('Copying %(source)s to %(destination)s'),
+                                {'source': os.path.join(bag_dir, f), 'destination': new_f})
+                    if os.path.isdir(os.path.join(bag_dir, f)):
+                        shutil.copytree(os.path.join(bag_dir, f), new_f)
+                    else:
+                        shutil.copy(os.path.join(bag_dir, f), new_f)
+                else:
+                    LOGGER.info(_('Moving %(source)s to %(destination)s'),
+                                {'source': os.path.join(bag_dir, f), 'destination': new_f})
+                    os.rename(os.path.join(bag_dir, f), new_f)
 
-            LOGGER.info(_('Moving %(source)s to %(destination)s'),
-                        {'source': temp_data, 'destination': 'data'})
-            os.rename(temp_data, 'data')
+            if data_subdir:
+                LOGGER.info(_('Moving %(source)s to %(destination)s'),
+                            {'source': temp_data, 
+                                'destination': os.path.join(data_dir, data_subdir)})
+                if not os.path.exists(data_dir):
+                    os.mkdir(data_dir)
+                elif os.path.exists(os.path.join(data_dir, data_subdir)):
+                    shutil.rmtree(os.path.join(data_dir, data_subdir))
+                os.rename(temp_data, os.path.join(data_dir, data_subdir))
+            else:
+                LOGGER.info(_('Moving %(source)s to %(destination)s'),
+                            {'source': temp_data, 'destination': data_dir})
+                os.rename(temp_data, data_dir)
 
             # permissions for the payload directory should match those of the
             # original directory
-            os.chmod('data', os.stat(cwd).st_mode)
+            os.chmod(data_dir, os.stat(bag_dir).st_mode)
+
+            if dest_dir:
```

Comment:
I'm wondering whether this should be one call with `src_dir` either being `bag_dir` or `None` for clarity